receiver/pluginreceiver/plugin.go

```diff
@@ -91,6 +100,32 @@ func (p *Plugin) ApplyDefaults(values map[string]any) map[string]any {
 	return result
 }
 
+// CheckExtensions checks to see if there exists any filestorage extensions
+// within the plugin config. If one exists, make a unique storage extension
+// directory based on the receiver component ID.
+func CheckExtensions(values map[string]any, id string) error {
+	for i, ext := range values {
+		if i == "file_storage" {
+			var cfg filestorage.Config
+			err := mapstructure.Decode(ext, &cfg)
+			if err != nil {
+				return err
+			}
+
+			name := strings.Replace(id, "/", "_", -1)
+			cfg.Directory = filepath.Join(cfg.Directory, name)
+			values[i] = map[string]any{"directory": cfg.Directory}
+			if _, err := os.Stat(cfg.Directory); os.IsNotExist(err) {
+				err = os.Mkdir(cfg.Directory, 0750)
+				if err != nil {
+					return err
+				}
+			}
```

Comment:
I don't think you need to stat I think you can just do a [os.MkdirAll](https://pkg.go.dev/os#MkdirAll) here and if it already exist it's a noop.